packages/auth0-api-js/src/api-client.ts

```
@@ -97,11 +199,146 @@ export class ApiClient {
         algorithms: ['RS256'],
         requiredClaims: ['iat', 'exp', ...(options.requiredClaims || [])],
       });
+
+      let cnfJkt: string | undefined;
+      const cnf = (payload as Record<string, unknown> & { cnf?: unknown }).cnf;
+
+      // Extract `jkt` from `cnf` claim if present
+      if (cnf && typeof cnf === 'object') {
+        const maybeJkt = (cnf as Record<string, unknown>).jkt;
+        if (typeof maybeJkt === 'string') {
+          cnfJkt = maybeJkt;
+        }
+      }
+
+      const hasProof = typeof dpopProof === 'string';
+
+      // DPoP validation logic
+      if (mode !== 'disabled' && scheme === 'bearer' && hasProof && !cnfJkt) {
+        throw this.#addChallenges(
+          new InvalidRequestError('DPoP proof requires the DPoP authentication scheme, not Bearer'),
+          mode,
+          scheme
+        );
+      }
+
+      // Determine if DPoP verification is needed
+      const shouldVerifyDpop =
+        mode !== 'disabled' && (mode === 'required' || scheme === 'dpop' || hasProof || !!cnfJkt);
+
+      // Enforce DPoP binding when `cnf.jkt`
+      if (mode !== 'disabled' && scheme === 'dpop' && !cnfJkt) {
+        throw this.#addChallenges(
+          new DpopBindingMismatchError('JWT Access Token has no jkt confirmation claim'),
+          mode,
+          scheme,
+          {
+            dpopSpecific: true,
+          }
+        );
+      }
```

This should be throw `invalid_token` error code for the Allowed Mode.